tests/utils/index.js

```diff
@@ -1115,32 +1116,38 @@ const waitForDockerLogs = (container, ...regex) => {
   let receivedFirstLine;
   const firstLineReceivedPromise = new Promise(resolve => receivedFirstLine = resolve);
 
+  const checkOutput = (data) => {
+    if (!firstLine) {
+      firstLine = true;
+      receivedFirstLine();
+      return;
+    }
+
+    data = data.toString();
+    logs += data;
+    const lines = data.split('\n');
+    const matchingLine = lines.find(line => regex.find(r => r.test(line)));
+    return matchingLine;
+  };
+
   const promise = new Promise((resolve, reject) => {
     timeout = setTimeout(() => {
       console.log('Found logs', logs, 'did not match expected regex:', ...regex);
       reject(new Error('Timed out looking for details in logs.'));
       killSpawnedProcess(proc);
     }, 20000);
 
-    const checkOutput = (data) => {
-      if (!firstLine) {
-        firstLine = true;
-        receivedFirstLine();
-        return;
-      }
-
-      data = data.toString();
-      logs += data;
-      const lines = data.split('\n');
-      if (lines.find(line => regex.find(r => r.test(line)))) {
+    const check = data => {
+      const foundMatch = checkOutput(data);
+      if (foundMatch) {
         resolve();
         clearTimeout(timeout);
         killSpawnedProcess(proc);
       }
     };
 
-    proc.stdout.on('data', checkOutput);
-    proc.stderr.on('data', checkOutput);
+    proc.stdout.on('data', check);
+    proc.stderr.on('data', check);
```

Comment:
This is just a refactor right? Did Sonar get you?

Reply:
Correct. It complained about code nesting depth.